src/mbgl/util/string_indexer.cpp

```diff
@@ -49,7 +38,10 @@ StringIdentity StringIndexer::get(const std::string& string) {
         auto result = stringToIdentity.insert({string, id});
         if (result.second) {
             // this writer made the insert
-            identityToString.push_back(string);
+            identityToString.push_back(std::string(string));
+            stringToIdentity.erase(result.first);
+            result = stringToIdentity.insert({std::string_view(identityToString.back().data()), id});
```

Comment:
I could be missing something, but it seems like `find` would be better than `insert` and `erase`.

And really minor, but `string_view(const char*)` does a `strlen`, using the overload that takes the already-known length would be ever so slightly faster.